x/erc20/keeper/ibc_callbacks.go

```diff
@@ -111,28 +122,47 @@ func (k Keeper) OnRecvPacket(
 }
 
 // OnAcknowledgementPacket responds to the the success or failure of a packet
-// acknowledgement written on the receiving chain. If the acknowledgement
-// was a success then nothing occurs. If the acknowledgement failed, then
-// the sender is refunded and then the IBC Coins are converted to ERC20.
+// acknowledgement written on the receiving chain. If the acknowledgement was a
+// success then nothing occurs. If the acknowledgement failed, then the sender
+// is refunded and then the IBC Coins are converted to ERC20.
 func (k Keeper) OnAcknowledgementPacket(
 	ctx sdk.Context, _ channeltypes.Packet,
 	data transfertypes.FungibleTokenPacketData,
 	ack channeltypes.Acknowledgement,
 ) error {
 	switch ack.Response.(type) {
 	case *channeltypes.Acknowledgement_Error:
+		sender, err := sdk.AccAddressFromBech32(data.GetSender())
+		if err != nil {
+			return err
+		}
+
+		// assume that all module accounts on Evmos need to have their tokens in the
+		// IBC representation as opposed to ERC20
+		if types.IsModuleAccount(ctx, k.accountKeeper, sender) {
+			return nil
+		}
 		// convert the token from Cosmos Coin to its ERC20 representation
 		return k.ConvertCoinToERC20FromPacket(ctx, data)
 	default:
-		// the acknowledgement succeeded on the receiving chain so nothing
-		// needs to be executed and no error needs to be returned
+		// the acknowledgement succeeded on the receiving chain so nothing needs to
+		// be executed and no error needs to be returned
 		return nil
 	}
 }
 
 // OnTimeoutPacket converts the IBC coin to ERC20 after refunding the sender
 // since the original packet sent was never received and has been timed out.
 func (k Keeper) OnTimeoutPacket(ctx sdk.Context, _ channeltypes.Packet, data transfertypes.FungibleTokenPacketData) error {
+	sender, err := sdk.AccAddressFromBech32(data.Sender)
+	if err != nil {
+		return err
+	}
+
+	// assume that all module accounts on Evmos need to have their tokens in the IBC representation as opposed to ERC20
+	if types.IsModuleAccount(ctx, k.accountKeeper, sender) {
```

Comment:
actually, let's move this to `ConvertCoinToERC20FromPacket` to avoid duplication